### PR TITLE
Halfedge Selection

### PR DESCRIPTION
**Pull request by ptpham**
Definitely didn't get as far as I wanted to. XD I'll get to viewable layerings this week.

### CODE REVIEW

File: lib/alignment.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      }
+
+      if (!overlap) result.push(+level);
+    }
+    return result;
+  }
+}
+
+var _v3_0 = vec3.create();
+function reflect(points, begin, end) {
```
Comment (cixzhang) [inline lib/alignment.js]: Should this be in geometry?

Comment (ptpham) [inline lib/alignment.js]: Uh ... well I put the main logic in linalg3. This is sort of a wrapper that does some preprocessing. I'd rather leave it here for now if you're not too opposed. XD

File: lib/linalg3.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
 
-var { mat3 } = require('gl-matrix');
+var { vec3, mat3 } = require('gl-matrix');
 var EPSILON = require('./geometry').EPSILON;
 
+var _v3_0 = vec3.create();
+var _v3_1 = vec3.create();
```
Comment (ptpham) [inline lib/linalg3.js]: Addressed.

File: lib/linalg3.js [hunk 3]
```diff
@@ -49,5 +53,15 @@ function polarUP(U, P, A, epsilon) {
   return U;
 }
 
-module.exports = { polarU, polarUP };
+function reflectAround(result, point, base, dir) {
```
Comment (ptpham) [inline lib/linalg3.js]: Addressed

**Pull request merged**